src/main/java/io/r2dbc/postgresql/message/backend/CopyData.java

```diff
@@ -77,7 +77,7 @@ public String toString() {
     static CopyData decode(ByteBuf in) {
         Assert.requireNonNull(in, "in must not be null");
 
-        return new CopyData(in.readSlice(in.readableBytes()));
+        return new CopyData(in.readSlice(in.readableBytes()).retain());
```

Comment:
Actually it's a leak since CopyData has no release method, but I'm not sure if I should fix that in this pull request.